NEWS.md

```diff
@@ -292,6 +292,78 @@
     # 2:     2    10
     ```
 
+40. `first()` and `last()` gain `na.rm` taking values `FALSE` (default), `TRUE` or `"row"`, [#4239](https://github.com/Rdatatable/data.table/issues/4239). For vector input, `TRUE` and `"row"` are the same. For `data.table|frame` input, `TRUE` returns the first/last non-NA observation in each column, while `"row"` returns the first/last row where all columns are non-NA. `TRUE` is optimized by group and `"row"` may be optimized by group in future. `n>1` with `na.rm=TRUE` is also optimized by group. Thanks to Nicolas Bennett and Michael Chirico for the requests, and Benjamin Schwendinger for the PR.
```

Comment:
> For vector input

Since R considers `list(NA, 1, 2, NA)` to be a "vector", can we clarify any relevant difference for list vs. data.frame input?

Reply:
If we add an option for `"row"`, it would somewhat feel natural to me to also include an option `"col"` which has the same behavior as `TRUE`.

Reply:
I'm not sure what you mean here. I don't think readers will confuse data.frame when they see 'vector input'. After all `is.vector` is false for data.frame. `list(NA,1,2,NA)` is a vector yes and is intended to be referred to by 'for vector input'.
Is there a wording change you can suggest since I'm not following?

Reply:
@ben-schwen Yes that crossed my mind too. But then I thought some users might think "col" would mean removing the column?  Then I thought maybe "within-col" or "by-col" and I couldn't come up with anything that clicked. Trying to think of a term that conveys removing NA's within each column.  Then what about when a vector is the input and na.rm='col' is specified: should it still work, or warn/error that data.frame|table input is expected when na.rm='col'?

Reply:
True and "byrow" and "bycol" are more clumsy while not bringing much to the table. Maybe warn when "row" and "col" are used with vector input but still proceed as with `TRUE`?

Reply:
re: my comment, I had written that before reading the full PR. Makes more sense now, please proceed (not resolving b/c of the parallel discussion on `"row`"/`"col"`

Reply:
resolved for now (I hope) by adding explanation to ?last and asking for feedback: https://github.com/Rdatatable/data.table/pull/5168/commits/31accc5ba04191a8268dd1c8369e08d4d8efc7cc.